UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -811,10 +722,10 @@ void ResetIfDone()
         void SendInfo()
         {
             // If the Agent is done, it has just reset and thus requires a new decision
-            if (m_RequestDecision || IsDone())
+            if (m_RequestDecision || m_Done)
             {
                 SendInfoToBrain();
-                ResetReward();
+                m_Reward = 0f;
```

Comment:
Need to reset m_CumulativeReward too?

Reply:
No, this is called every step

Reply:
ResetReward() would reset m_CumulativeReward if m_Done was true.

Reply:
I think this is handled through some reset logic now.